src/rust/engine/fs/store/src/local.rs

```diff
@@ -57,89 +73,52 @@ impl ByteStore {
     &self.inner.executor
   }
 
-  // Note: This performs IO on the calling thread. Hopefully the IO is small enough not to matter.
-  pub fn entry_type(&self, fingerprint: &Fingerprint) -> Result<Option<EntryType>, String> {
-    if *fingerprint == EMPTY_DIGEST.0 {
+  pub async fn entry_type(&self, fingerprint: Fingerprint) -> Result<Option<EntryType>, String> {
+    if fingerprint == EMPTY_DIGEST.0 {
       // Technically this is valid as both; choose Directory in case a caller is checking whether
       // it _can_ be a Directory.
       return Ok(Some(EntryType::Directory));
     }
-    let effective_key = VersionedFingerprint::new(*fingerprint, ShardedLmdb::schema_version());
-    {
-      let (env, directory_database, _) = self.inner.directory_dbs.clone()?.get(fingerprint);
-      let txn = env
-        .begin_ro_txn()
-        .map_err(|err| format!("Failed to begin read transaction: {:?}", err))?;
-      match txn.get(directory_database, &effective_key) {
-        Ok(_) => return Ok(Some(EntryType::Directory)),
-        Err(NotFound) => {}
-        Err(err) => {
-          return Err(format!(
-            "Error reading from store when determining type of fingerprint {}: {}",
-            fingerprint, err
-          ));
-        }
-      };
+
+    // In parallel, check for the given fingerprint in both databases.
+    let d_dbs = self.inner.directory_dbs.clone()?;
+    let is_dir = d_dbs.exists(fingerprint);
+    let f_dbs = self.inner.file_dbs.clone()?;
+    let is_file = f_dbs.exists(fingerprint);
+
+    // TODO: Could technically use select to return slightly more quickly with the first
```

Comment:
Is the intent to fix this TODO at some point?

Reply:
Possibly? It's preeeetty micro-optimizey, but might be worth doing the next time someone is in there.

Reply:
My concern is that this introduces a TODO that will never get done. Maybe just have the comment not say TODO if there is no intent to change later?